src/client.ts

```diff
@@ -765,13 +780,18 @@ export class StreamChat<
       this._logApiError(type, url, e);
 
       if (e.response) {
-        if (
-          e.response.data.code === chatCodes.TOKEN_EXPIRED &&
-          !this.tokenManager.isStatic()
-        ) {
+        const code = e.response.data && e.response.data.code;
+        const headers = e.response.headers;
+
+        if (code === chatCodes.TOKEN_EXPIRED && !this.tokenManager.isStatic()) {
           this.tokenManager.loadToken();
           return await this.doAxiosRequest<T>(type, url, data, options);
         }
+
+        if (code === chatCodes.RATE_LIMITED && headers && headers['x-ratelimit-reset']) {
+          this.rateLimitResetTime[url] = e.response.headers['x-ratelimit-reset'];
```

Comment:
is `url` a reliable key to cache the rate limits here? mostly because we have rate limits per endpoints and this is an enriched url.
I think either we go with a rate limit mapping to functions which could be lot of work or just cache based on function names, also a bit of work

Reply:
Yeah I realized that as well. Thinking of better way to approach it somehow:

1. Mapping based on handler names
2. Give access to some kind of callback function to user when rate limit is hit